skrf/network.py

```diff
@@ -6403,6 +6418,74 @@ def innerconnect_s(A: np.ndarray, k: int, l: int) -> np.ndarray:
 
     return C
 
+def innerconnect_s_lstsq(A: np.ndarray, k: int, l: int) -> np.ndarray:
+    """
+    Connect two ports of a single n-port network's s-matrix using a
+    least-squares solution. It uses a least-squares approach to handle
+    cases where the determinant of the sub-matrix is close to zero, which
+    can lead to numerical instability in the direct formula.
+
+
+    Note
+    ----
+    If the determinant of the sub-matrix is not close to zero, it is recommended
+    to use the `innerconnect_s` function instead, which uses a direct formula
+    for better numerical stability.
+
+
+    Parameters
+    ----------
+    A : :class:`numpy.ndarray`
+        S-parameter matrix of `A`, shape is fxnxn
+    k : int
+        port index on `A` (port indices start from 0)
+    l : int
+        port index on `A`
+
+    Returns
+    -------
+    AEE : :class:`numpy.ndarray`
+            new S-parameter matrix
+
+
+    See Also
+    --------
+    connect_s : actual  S-parameter connection algorithm.
+    innerconnect_s : actual S-parameter connection algorithm.
+    """
+    warnings.warn(
+        'Singular matrix detected, using numpy.linalg.lstsq instead.',
+        RuntimeWarning,
+        stacklevel=2
+    )
```

Comment:
Since `innerconnect_s_lstsq()` is automatically called within `innerconnect_s()` when `det=0`, rather than being manually invoked by the user.
It is necessary to inform the user through a `warning` or `other methods` that the result of the `Networks` connectionn is obtained through least squares method rather than an exact theoretical calculation.

Reply:
Apologies, my mistake.